stdlib/Dates/src/exceptions.jl

```diff
@@ -0,0 +1,9 @@
+struct UndefDateFormatCode <: Exception
```

Comment:
Just a quick check.
Right now if open a repl and run `subtypes(Exception)`
there are 61 types shown.
The only ones that don't end with `Exception` or `Error` are:
`SegmentationFault`, and `DimensionMismatch`

Should we instead name this
```suggestion
struct DateFormatCodeUndefinedException <: Exception
```